lib/jellyfish/event.ex

```diff
@@ -67,4 +106,31 @@ defmodule Jellyfish.Event do
 
   defp to_proto_server_notification({:hls_upload_crashed, room_id}),
     do: {:hls_upload_crashed, %HlsUploadCrashed{room_id: room_id}}
+
+  defp to_proto_track(%Jellyfish.Track{} = track) do
+    %Track{
+      id: track.id,
+      type: to_proto_track_type(track.type),
+      encoding: to_proto_encoding(track.encoding),
+      metadata: Jason.encode!(track.metadata)
+    }
+  end
+
+  defp to_proto_track(%Message.TrackAdded{} = track) do
+    %Track{
+      id: track.track_id,
+      type: to_proto_track_type(track.track_type),
+      encoding: to_proto_encoding(track.track_encoding),
+      metadata: Jason.encode!(track.track_metadata)
+    }
+  end
+
+  defp to_proto_encoding(:H264), do: :ENCODING_H264
+  defp to_proto_encoding(:VP8), do: :ENCODING_VP8
+  defp to_proto_encoding(:OPUS), do: :ENCODING_OPUS
+  defp to_proto_encoding(_encoding), do: :ENCODING_UNSPECIFIED
```

Comment:
This is created from event sent from Rtc Engine, so I guess its better to have track with unknown encoding, than crashing Jellyfish.
Also, as you proposed I will remove the encoding from the `Track` anyway?

Reply:
I am not sure about that. I think that if a jellyfish doesn't expect something from the engine, it should raise an error. I think it would be more aligned with the BEAM approach, `Let it crash.`